packages/millicast-sdk/src/utils/Diagnostics.js

```diff
@@ -28,21 +35,39 @@ const setStats = (stats) => {
   diagnostics.stats.push(stats)
 }
 
-const getDiagnostics = (statsCount = 5) => {
+const initStats = async (statsCount) => {
+  try {
+    webRTCPeer.initStats()
+    await new Promise(resolve => setTimeout(resolve, statsCount * 1010))
```

Comment:
I advice we'd better to listen to the event of stats instead of using a timer here

Reply:
After looked further, I have advices below:
1. this function named `initStats` but it stops Stats after a while by the timer, this will confuse the user, we could provide a `stopStats` function for user
2. I see we have max stats number protection when we record the stats in the callback, so we don't need the count down timer here and we will always get the latest stats and remove the deprecated ones.
3. will the stopStats conflict with user's initStats, I mean user started the stats but have not stopped it yet?

Reply:
1. I agree the function does not give full visibility, maybe `getStats` would be better. Anyway, this is not an exposed function to the user, so the user should not be able to use it.
2. and 3. This function is only called when there are no stats available, meaning that the `initStats` was never called, so the timer will be based on how many stats are asked for. Something I will add is to check that the count passed is not bigger than the maximum.

Reply:
Regarding question 3,  will our `stopStats` lead to unexpected behaviour when user want to keep receiving the stats?

And since we only save the latest N stats records, I don't think we need the timer and call `stopStats`.  I am thinking we should have a global switch for diagnostics, when it's on, we should keep stats open even after user called `stopStats`. But when diagnostics off, it will depends on user's call to turn on/off stats.